dashboards-observability/common/constants/autocomplete.ts

```diff
@@ -67,27 +67,27 @@ export interface DataItem {
   doc_count: any;
 }
 
-const JUST_SEARCH_REGEX = /\s*search\s+source\s*=\s*(\S+)/;
-const SEARCH_WHERE_REGEX = /\s*search\s+source\s*=\s*(\S+)\s*\|\s*where\s+\S+\s*=\s*\S+/;
-const SEARCH_MATCH_REGEX = /\s*search\s+source\s*=\s*(\S+)\s*\|\s*where\s+match\(\S+,\s*\S+\)/;
-const JUST_SOURCE_REGEX = /\s*source\s*=\s*(\S+)/;
-const SOURCE_WHERE_REGEX = /\s*source\s*=\s*(\S+)\s*\|\s*where\s+\S+\s*=\s*\S+/;
-const SOURCE_MATCH_REGEX = /\s*source\s*=\s*(\S+)\s*\|\s*where\s+match\(\S+,\s*\S+\)/;
+const JUST_SEARCH_REGEX = /\s*search\s+source\s*=\s*[^\\\/\?\"\<\>\|\s\,\#]*(\s*,\s*[^\\\/\?\"\<\>\|\s\,\#]+)*/;
```

Comment:
> `\s*[^\\\/\?\"\<\>\|\s\,\#]*`

the first `\s*` is redundant? also could you explain a bit like why punctuations will appear after `source = `?